src/app/components/queue/form/QuestionForm.tsx

```diff
@@ -0,0 +1,209 @@
+import {
+  Box,
+  Button,
+  Checkbox,
+  Drawer,
+  FormControl,
+  FormControlLabel,
+  FormLabel,
+  TextField,
+} from "@mui/material";
+import { IoMdClose } from "react-icons/io";
+import React from "react";
+import { officeHourContext } from "@context/OfficeHourContext";
+import { MultipleChoiceTags, SingleChoiceTags } from "./Tags";
+import { TagOption } from "@interfaces/db";
+import { createQuestion } from "@utils/index";
+import { serverTimestamp } from "firebase/firestore";
+import { UserSessionContext } from "@context/UserSessionContext";
+import { IdentifiableQuestion } from "@interfaces/type";
+
+interface QuestionFormProps {
+  // button to open the form
+  triggerButton: JSX.Element;
+  title: string;
+  // current state of question
+  currentQuestion: IdentifiableQuestion;
+}
+
+const QuestionForm = (props: QuestionFormProps) => {
+  const { triggerButton, title, currentQuestion } = props;
+  const [openForm, setOpenForm] = React.useState(false);
+  const ohContext = React.useContext(officeHourContext);
+  const userSessionContext = React.useContext(UserSessionContext);
+
+  const [question, setQuestion] = React.useState(currentQuestion.title);
+  const [description, setDescription] = React.useState(
+    currentQuestion.description
+  );
+  const [questionPublic, setQuestionPublic] = React.useState(
+    currentQuestion.questionPublic
+  );
+
+  const [questionTags, setQuestionTags] = React.useState<
```

Comment:
Spoke offline, we should derive `questionTags` from `currentQuestion`.

Reply:
I am following the logic, and it should populate the questionTags for either the "create new form" and "edit form". It is a little tricky though, I'm not sure if we can improve it for our types.